crates/ruff_python_formatter/src/verbatim.rs

```diff
@@ -573,33 +596,275 @@ impl Format<PyFormatContext<'_>> for TrailingFormatOffComment<'_> {
     }
 }
 
-struct VerbatimText(TextRange);
+/// Stores the indentation of a statement by storing the number of indentation characters.
+/// Storing the number of indentation characters is sufficient because:
+/// * Two indentations are equal if they result in the same column, regardless of the used tab size.
+///   This implementation makes use of this fact and assumes a tab size of 1.
+/// * The source document is correctly indented because it is valid Python code (or the formatter would have failed parsing the code).
+#[derive(Copy, Clone)]
+struct Indentation(u32);
+
+impl Indentation {
+    fn from_stmt(stmt: &Stmt, source: &str) -> Indentation {
+        let line_start = Locator::new(source).line_start(stmt.start());
```

Comment:
How does this implementation handle multi-statement lines? Like `x = 1; import os`. Or, is that not relevant since you can't have a comment _within_ a multi-statement line (and so the `first_suppressed` is guaranteed to be at the start of a logical line)?

Reply:
This is an interesting case. This works because the `fmt: off` comment must be an own line comment. Meaning either the whole multiline statement gets formatted or suppressed.